homeassistant/components/azure_event_hub/__init__.py

```diff
@@ -1,89 +1,213 @@
+from azure.eventhub import (
+    EventData,
+    EventDataBatch,
+    EventHubProducerClient,
+    EventHubSharedKeyCredential,
+)
+from azure.eventhub.exceptions import EventHubError
 import voluptuous as vol
 
 from homeassistant.const import (
     EVENT_HOMEASSISTANT_STOP,
-    EVENT_STATE_CHANGED,
+    MATCH_ALL,
     STATE_UNAVAILABLE,
     STATE_UNKNOWN,
 )
-from homeassistant.core import Event, HomeAssistant
+from homeassistant.core import Event, HomeAssistant, callback
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entityfilter import FILTER_SCHEMA
 from homeassistant.helpers.json import JSONEncoder
 
-_LOGGER = logging.getLogger(__name__)
+from .const import (
+    CONF_EVENT_HUB_CON_STRING,
+    CONF_EVENT_HUB_INSTANCE_NAME,
+    CONF_EVENT_HUB_NAMESPACE,
+    CONF_EVENT_HUB_SAS_KEY,
+    CONF_EVENT_HUB_SAS_POLICY,
+    CONF_FILTER,
+    DOMAIN,
+)
 
-DOMAIN = "azure_event_hub"
+_LOGGER = logging.getLogger(__name__)
 
-CONF_EVENT_HUB_NAMESPACE = "event_hub_namespace"
-CONF_EVENT_HUB_INSTANCE_NAME = "event_hub_instance_name"
-CONF_EVENT_HUB_SAS_POLICY = "event_hub_sas_policy"
-CONF_EVENT_HUB_SAS_KEY = "event_hub_sas_key"
-CONF_FILTER = "filter"
+QUEUE_BACKLOG_SECONDS = 30
+PAUSE_WHEN_EMPTY = 5
+BATCH_TIMEOUT = 1
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
-                vol.Required(CONF_EVENT_HUB_NAMESPACE): cv.string,
-                vol.Required(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
-                vol.Required(CONF_EVENT_HUB_SAS_POLICY): cv.string,
-                vol.Required(CONF_EVENT_HUB_SAS_KEY): cv.string,
-                vol.Required(CONF_FILTER): FILTER_SCHEMA,
-            }
+                vol.Optional(CONF_EVENT_HUB_CON_STRING): cv.string,
+                vol.Optional(CONF_EVENT_HUB_NAMESPACE): cv.string,
+                vol.Optional(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
+                vol.Optional(CONF_EVENT_HUB_SAS_POLICY): cv.string,
+                vol.Optional(CONF_EVENT_HUB_SAS_KEY): cv.string,
+                vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+            },
+            cv.has_at_least_one_key(
+                CONF_EVENT_HUB_CON_STRING, CONF_EVENT_HUB_NAMESPACE
+            ),
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-async def async_setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
+async def async_setup(hass, yaml_config):
     """Activate Azure EH component."""
     config = yaml_config[DOMAIN]
-
-    event_hub_address = (
-        f"amqps://{config[CONF_EVENT_HUB_NAMESPACE]}"
-        f".servicebus.windows.net/{config[CONF_EVENT_HUB_INSTANCE_NAME]}"
-    )
-    entities_filter = config[CONF_FILTER]
-
-    client = EventHubClientAsync(
-        event_hub_address,
-        debug=True,
-        username=config[CONF_EVENT_HUB_SAS_POLICY],
-        password=config[CONF_EVENT_HUB_SAS_KEY],
+    entities_filter = config.get(CONF_FILTER)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.